compiler/rustc_parse/src/parser/pat.rs

```diff
@@ -789,7 +789,7 @@ impl<'a> Parser<'a> {
 
     fn error_inclusive_range_match_arrow(&self, span: Span) {
         let after_pat = span.with_hi(span.hi() - rustc_span::BytePos(1)).shrink_to_hi();
-        self.sess.emit_err(InclusiveRangeMatchArrow { span, after_pat });
+        self.sess.emit_note(InclusiveRangeMatchArrow { span, after_pat });
```

Comment:
I think this should still be an error, and keep the primary message that explains the error. 

The "this is parsed as an inclusive range" should just be a note (or probably a *label*) appended to this error.